### PR TITLE
Test multiple instance configuration in mariadb

### PR DESCRIPTION
**Pull request by punkioudi**
The test is for now excluded from `Tumbleweed ppc64le` and `ppc64`. The issue is tracked by a [progress ticket ](https://progress.opensuse.org/issues/99558) and will be resolved in a future PR.

- Related ticket: https://progress.opensuse.org/issues/98256
- Needles: N/A
- Verification run: 
SLES [15sp3](https://openqa.suse.de/tests/7275418#step/mariadb_srv/41) | [15sp2](https://openqa.suse.de/tests/7275420#step/mariadb_srv/41) | [15sp1](https://openqa.suse.de/tests/7275421#step/mariadb_srv/41) | [15](https://openqa.suse.de/tests/7275422#step/mariadb_srv/41) | [12sp5](https://openqa.suse.de/tests/7275424#step/mariadb_srv/41) | [12sp4](https://openqa.suse.de/tests/7275425#step/mariadb_srv/41) | [12sp3](https://openqa.suse.de/tests/7275435#step/mariadb_srv/13) | [12sp2](https://openqa.suse.de/tests/7275463#step/mariadb_srv/13)
Tumbleweed : [x86_64](https://openqa.opensuse.org/tests/1947019#step/mariadb_srv/41) | [aarch64](https://openqa.opensuse.org/tests/1947020#step/mariadb_srv/41) | [JeOS-for-RPi-aarch64](https://openqa.opensuse.org/tests/1947023#step/mariadb_srv/40) | [JeOS-for-kvm-and-xen-x86_64](https://openqa.opensuse.org/tests/1947026#step/mariadb_srv/41) | 
[Leap 15.3](https://openqa.opensuse.org/tests/1947134) | [Leap 15.2](https://openqa.opensuse.org/tests/1947025#step/mariadb_srv/41)

### CODE REVIEW

File: tests/console/mariadb_srv.pm [hunk 1]
```diff
@@ -35,6 +36,23 @@ sub run {
...
+    assert_script_run "touch /etc/mynode2.cnf";
+    assert_script_run "curl " . data_url('console/mariadb/mynode1.cnf') . " -o /etc/mynode1.cnf";
+    assert_script_run "curl " . data_url('console/mariadb/mynode2.cnf') . " -o /etc/mynode2.cnf";
+    assert_script_run "mkdir -p /var/lib/mysql/node1";
+    assert_script_run "mkdir -p /var/lib/mysql/node2";
+    assert_script_run "chown mysql:root /var/lib/mysql/node1";
+    assert_script_run "chown mysql:root /var/lib/mysql/node2";
+
+    systemctl 'start mariadb@node1.service';
+    systemctl 'start mariadb@node2.service';
```
Comment (foursixnine) [inline tests/console/mariadb_srv.pm:51]: Since you're starting two nodes, it would make a lot of sense to also bring down the nodes after the test is done, just make sure it's also called in the post_fail_hook just in case the test fails..

**Comment by michaelgrifalconi** [comment]
Great work! :) One question, does it need root-console or could it work with the serial one as well?

**Comment by dzedro** [comment]
> Great work! :) One question, does it need root-console or could it work with the serial one as well?

Console tests should use serial console.

**Comment by punkioudi** [comment]
> Great work! :) One question, does it need root-console or could it work with the serial one as well?

There is the `assert_screen 'test-mysql_srv-1';` from the first test which I didn't change and it needs `root-console`. Probably yeah it could be changed, by greping the items that it is looking for in the screenshot :)

**Comment by dzedro** [comment]
> > > Great work! :) One question, does it need root-console or could it work with the serial one as well?
> > 
> > 
> > Console tests should use serial console.
> 
> Console is changed now ;) I also removed the `assert_screen`

Great, always avoid use of needles in console. You can run `systemctl is-active`

File: tests/console/mariadb_srv.pm [hunk 2]
```diff
@@ -33,8 +42,35 @@ sub run {
...
     systemctl "start $mariadb";
-    systemctl "status $mariadb";
-    assert_screen 'test-mysql_srv-1';
+    systemctl "is-active $mariadb";
+
+    # Test multiple instance configuration
+    # It is not supported in sle12sp2 and sle12sp3
+    if (!is_sle('<=12-SP3') && !is_ppc64le && !is_ppc64) {
+        assert_script_run "touch /etc/mynode1.cnf";
+        assert_script_run "touch /etc/mynode2.cnf";
```
Comment (dzedro) [inline tests/console/mariadb_srv.pm:51]: You could do all this same commands with [brace expansion](https://www.gnu.org/software/bash/manual/html_node/Brace-Expansion.html) e.g. ` mkdir -p /var/lib/mysql/node{1,2}`

Comment (punkioudi) [reply tests/console/mariadb_srv.pm]: I kept only the `systemctl` commands as same for visibility ;)

File: lib/Utils/Architectures.pm [hunk 3]
```diff
@@ -128,6 +129,17 @@ sub is_ppc64le {
     return check_var('ARCH', 'ppc64le');
 }
 
+=head2 is_ppc64
```
Comment (foursixnine) [inline lib/Utils/Architectures.pm:132]: unrelated change?

Comment (punkioudi) [reply lib/Utils/Architectures.pm:132]: As I write in the description,I excluded this test from ppc64 arch in Tumbleweed. So it is not unrelated, as I added this in `Architectures.pm` to keep the same code style here : `if (!is_sle('<=12-SP3') && !is_ppc64le && !is_ppc64)`

Comment (punkioudi) [reply lib/Utils/Architectures.pm:132]: As it is related, do you still think that I have to add it in a different PR or commit?

Comment (foursixnine) [reply lib/Utils/Architectures.pm:132]: @punkioudi ~~I don't see the changes in the PR that use `is_ppc64` :)~~ Nope, forget what I wrote :)

**Review by foursixnine — APPROVED** [review]

**Review by dzedro — APPROVED** [review]
